libs/ng-utils/src/lib/use-http-image-source/use-http-image-source.pipe.spec.ts

```diff
@@ -0,0 +1,110 @@
+import { HttpClientTestingModule, HttpTestingController } from '@angular/common/http/testing';
+import { ChangeDetectorRef } from '@angular/core';
+import { fakeAsync, TestBed, tick, waitForAsync } from '@angular/core/testing';
+import { UseHttpImageSourcePipeModule } from './use-http-image-source-pipe.module';
+import { UseHttpImageSourcePipe } from './use-http-image-source.pipe';
+
+const MOCK_CDR = {
+  markForCheck: jest.fn(),
+};
+
+describe('UseHttpImageSourcePipe', () => {
+  let pipe: UseHttpImageSourcePipe;
+  let httpMock: HttpTestingController;
+
+  beforeEach(() => {
+    global.URL.createObjectURL = jest.fn().mockReturnValue('blob:testblob.png');
+    TestBed.configureTestingModule({
+      imports: [
+        HttpClientTestingModule,
+        UseHttpImageSourcePipeModule.forRoot({
+          loadingImagePath: 'test/loading.png',
+          errorImagePath: 'test/error.png',
+        }),
+      ],
+      providers: [
+        UseHttpImageSourcePipe,
+        {
+          provide: ChangeDetectorRef,
+          useValue: MOCK_CDR,
+        },
+      ],
+    }).compileComponents();
+
+    pipe = TestBed.inject(UseHttpImageSourcePipe);
+    httpMock = TestBed.inject(HttpTestingController);
+  });
+
+  afterEach(() => {
+    pipe.ngOnDestroy();
+    jest.resetAllMocks();
+  });
+
+  it(
+    `when receives a falsy value, it returns the error image path`,
+    waitForAsync(() => {
+      const result = pipe.transform('');
+      expect(result).toEqual('test/error.png');
+    })
+  );
+
+  it(`Initially it returns with the loading image path`, fakeAsync(() => {
+    const result = pipe.transform('test/something.png');
+    expect(result).toEqual('test/loading.png');
+  }));
+
+  it(`Sends a requests using the httpClient`, fakeAsync(() => {
+    const result = pipe.transform('test/something.png');
+    expect(result).toEqual('test/loading.png');
+
+    tick(100);
+    httpMock
+      .expectOne(`test/something.png`, 'should hit test/something.png')
+      .flush(new Blob(['something']));
+
+    tick(100);
+    expect(MOCK_CDR.markForCheck).toHaveBeenCalled();
+    const updatedResult = pipe.transform('test/something.png');
+    expect(updatedResult).toEqual({ changingThisBreaksApplicationSecurity: 'blob:testblob.png' });
```

Comment:
It took me a bit understanding where does this `changingThisBreaksApplicationSecurity` property comes from. We could get the `DomSanitizer` from injector and refactor this expect to:

```suggestion
    expect(updatedResult).toEqual(domSanitizer.bypassSecurityTrustUrl('blob:testblob.png'));
```

Reply:
okay, done